addons/web_editor/static/src/js/wysiwyg/widgets/image_crop.js

```diff
@@ -97,6 +97,59 @@ export class ImageCrop extends Component {
         }
     }
 
+    /**
+     * Crops image with the aspect ratio 1:1
+     */
+    async autoRatio(ratio, cropped) {
+        if (this.$cropperImage) {
+            this.$cropperImage.cropper("reset");
+            if (this.aspectRatio !== ratio) {
+                this.aspectRatio = ratio;
+                this.$cropperImage.cropper(
+                    "setAspectRatio",
+                    this.aspectRatios[this.aspectRatio].value
+                );
+            }
+            await this._save(cropped);
+        }
+    }
+
+    /**
+     * Crops image with the aspect ratio 1:1 for preview
+     */
+    async autoRatioPreview(ratio) {
+        if (this.$cropperImage) {
+            this.$cropperImage.cropper("reset");
+            if (this.aspectRatio !== ratio) {
+                this.$cropperImage.cropper("setAspectRatio", this.aspectRatios[ratio].value);
+            }
+            await this._save(false);
+        }
+    }
+
```

Comment:
Can we merge the functionality of `autoRatio` & `autoRatioPreview` in single function?

Reply:
Yes, we can write single function. i'm changing it and trying to make single function with least parameters.